app/models/experimentation/stale_patient_experiment.rb

```diff
@@ -8,37 +8,18 @@ class StalePatientExperiment < NotificationsExperiment
     # Eligible patients whose last visit was 35-365 days ago and
     # don't have an appointment in the future.
     def eligible_patients(date)
-      parameters = {
-        hypertension: "yes",
-        last_visit_since: (date + PATIENT_VISITED_SINCE).beginning_of_day,
-        last_visit_until: (date + PATIENT_VISITED_UNTIL).end_of_day,
-        no_appointments_after: date.end_of_day
-      }
-      sql = GitHub::SQL.new(<<~SQL, parameters)
-        SELECT DISTINCT patients.id
-        FROM patients
-        INNER JOIN medical_histories mh
-         ON patients.id = mh.patient_id
-        LEFT JOIN appointments on appointments.patient_id = patients.id
-          AND appointments.device_created_at BETWEEN :last_visit_since AND :last_visit_until
-        LEFT JOIN prescription_drugs on prescription_drugs.patient_id = patients.id
-          AND prescription_drugs.device_created_at BETWEEN :last_visit_since AND :last_visit_until
-        LEFT JOIN blood_sugars on blood_sugars.patient_id = patients.id
-          AND blood_sugars.device_created_at BETWEEN :last_visit_since AND :last_visit_until
-        LEFT JOIN blood_pressures on blood_pressures.patient_id = patients.id
-          AND blood_pressures.device_created_at BETWEEN :last_visit_since AND :last_visit_until
-        WHERE patients.deleted_at IS NULL
-          AND mh.hypertension = :hypertension
-          AND (appointments.id IS NOT NULL
-            OR prescription_drugs.id IS NOT NULL
-            OR blood_sugars.id IS NOT NULL
-            OR blood_pressures.id IS NOT NULL)
-          AND NOT EXISTS
-            (SELECT 1 FROM appointments WHERE appointments.patient_id = patients.id
-               AND appointments.scheduled_date >= :no_appointments_after)
-      SQL
+      current_month = date.beginning_of_month
+      last_visit_since = (date + PATIENT_VISITED_SINCE).beginning_of_day
+      last_visit_until = (date + PATIENT_VISITED_UNTIL).end_of_day
+      no_appointments_after = date.end_of_day
 
-      self.class.superclass.eligible_patients.where(id: sql.values)
+      self.class.superclass.eligible_patients
+        .joins("inner join reporting_patient_visits ON reporting_patient_visits.patient_id = patients.id")
+        .joins("left outer join appointments future_appointments on future_appointments.patient_id = patients.id
+                AND future_appointments.scheduled_date > '#{no_appointments_after}'")
+        .where(reporting_patient_visits: {month_date: current_month})
```

Comment:
[nit] This would be more expressive on the join.

Reply:
`.joins` doesn't have an option to bind strings like `where` can with `?`, which means you'd have to do a `#
'{current_month}'` :/ Hence the `where`.